### DISCUSSION TITLE
Unable to increase the number of conv layers per block

### DISCUSSION THREAD

**Discussion by Mousa-saleh** [neuropoly/axondeepseg#594] [Help needed] [answered]
Hello all,

I am trying to increase the number of conv layers from [2,2,2,2] to [3,3,3,3] but I am getting the following error.
![image](https://user-images.githubusercontent.com/85881282/150742428-5b7cd940-59f3-4acb-9022-d9fa6b4cb575.png)

Thanks for your help,

**Comment by hermancollin** [comment, marked as answer]
Hi @Mousa-saleh
Could you provide the other parameters you used? I'm mostly curious about these:
-depth
-size_of_convolutions_per_layer
-features_per_convolution

**Reply by Mousa-saleh** [reply]
Hi @hermancollin 

"depth": 4,  
  "convolution_per_layer": [3,3, 3, 2],  
  "size_of_convolutions_per_layer": [[5, 5], [3, 3], [3, 3], [3, 3]],  
  "features_per_convolution": [[[1, 16], [16,16],[16,16]], [[16, 32],[32,32],[32,32]],[[32,64], [64, 64],[64, 64]], [[64, 128], [128, 128]]],  

Thanks,

**Reply by hermancollin** [reply]
Hi @Mousa-saleh 
I was able to reproduce the error. The problem is your `size_of_convolutions_per_layer` list. The kernel sizes are given with a single integer. Since your first layer has 3 convolution blocks, you need to specify the size of the third block. For example, the following would work:
`"size_of_convolutions_per_layer": [[5, 5, 5], [3, 3, 3], [3, 3, 3], [3, 3]]`
